### ISSUE TITLE
Mix loops installing additional dependencies for vue()

### ISSUE THREAD

**Issue by DSpeichert** [JeffreyWay/laravel-mix#2686]
-   Laravel Mix Version: 6.0.0 (`npm list --depth=0`)
-   Node Version (`node -v`): v15.4.0
-   NPM Version (`npm -v`):6.14.8
-   OS: Arch Linux

### Description:
When adding `.vue()` in webpack.mix.js to enable Vue processing, laravel-mix loops attempting to install "additional dependencies"

```
[laravel@archlinux10 bug-repro]$ npm run prod

> @ prod /home/laravel/bug-repro
> npm run production


> @ production /home/laravel/bug-repro
> mix --production

        Additional dependencies must be installed. This will only take a moment.
 
        Running: npm install vue-template-compiler vue-loader@^15.9.5 --save-dev --legacy-peer-deps
 
        Okay, done. The following packages have been installed and saved to your package.json dependencies list:
 
        - vue-template-compiler
 
        - vue-loader@^15.9.5
 
        Finished. Please run Mix again.
 
[laravel@archlinux10 bug-repro]$ npm run prod

> @ prod /home/laravel/bug-repro
> npm run production


> @ production /home/laravel/bug-repro
> mix --production

        Additional dependencies must be installed. This will only take a moment.
 
        Running: npm install vue-template-compiler vue-loader@^15.9.5 --save-dev --legacy-peer-deps
 
        Okay, done. The following packages have been installed and saved to your package.json dependencies list:
 
        - vue-template-compiler
 
        - vue-loader@^15.9.5
 
        Finished. Please run Mix again.
 
[laravel@archlinux10 bug-repro]$ npm run prod

> @ prod /home/laravel/bug-repro
> npm run production


> @ production /home/laravel/bug-repro
> mix --production

        Additional dependencies must be installed. This will only take a moment.
 
        Running: npm install vue-template-compiler vue-loader@^15.9.5 --save-dev --legacy-peer-deps
 
        Okay, done. The following packages have been installed and saved to your package.json dependencies list:
 
        - vue-template-compiler
 
        - vue-loader@^15.9.5
 
        Finished. Please run Mix again.
```

### Steps To Reproduce:

```diff
mix.js('resources/js/app.js', 'public/js')
+.vue()
.extract()
.postCss('resources/css/app.css', 'public/css', [
    require('postcss-import'),
    require('tailwindcss'),
    require('autoprefixer'),
]);
```

Repo to reproduce: https://github.com/DSpeichert/bug-repro/tree/mix-loop

Actually running `npm install vue-template-compiler vue-loader@^15.9.5 --save-dev --legacy-peer-deps` manually fixes the loop.

**Comment by JeffreyWay** [comment]
I just tested your repo on a Mac, and everything worked correctly. Can anyone else confirm that this fails on Windows or Linux?

**Comment by peterthomson** [comment]
Yep. Just loops the same "Success" message:  	
```
Additional dependencies must be installed. This will only take a moment.
Running: npm install vue-loader@^15.9.5 --save-dev --legacy-peer-deps
Okay, done. The following packages have been installed and saved to your package.json dependencies list:
- vue-loader@^15.9.5
Finished. Please run Mix again.
```

**Comment by Krisell** [comment]
Same problem here, on `macOS 11.1` and `npm 6.14.10`, but fixed by manually running the dependency installation.

**Comment by Krisell** [comment]
I tested with the reproduction repo by @DSpeichert and I can confirm the issue also for `macOS 11.1` and `npm 6.14.10`

Note that the `vue-loader` and `vue-template-compiler` dependencies have been commited by mistake by @DSpeichert so to actually reproduce the issue, those must be removed first.

![image](https://user-images.githubusercontent.com/25909128/103133848-33f65400-46ad-11eb-9886-c82f65803a7a.png)

**Comment by DSpeichert** [comment]
Indeed, I've fixed my reproduction repository to remove the accidentally committed packages that @Krisell pointed out.

**Comment by thecrypticace** [comment]
So I can't exactly reproduce this but I have discovered a possibly related problem. Running mix with the production flags causes npm install to run with `NODE_ENV=production` which removes all dev dependencies.

Do you run into this if you just do `npm run mix` without the production flag?

**Comment by Krisell** [comment]
@thecrypticace I think that is indeed the problem. It works correctly when running `npx mix` (see image below), but when running `npm run production`, the additional dependencies are added to the `package.json` and `package-lock.json` files but no actual installation of these dependencies take places and they do not appear in `node_modules`.

![image](https://user-images.githubusercontent.com/25909128/103269339-ec760d80-49b5-11eb-9182-c11f1c783b9a.png)

Whereas running `npm run production` instead:
![image](https://user-images.githubusercontent.com/25909128/103269638-86d65100-49b6-11eb-899a-7cb3b712fb47.png)

**Comment by peterthomson** [comment]
It's not a great solution and I'm sure the core team will come up with a better fix. But if anyone just wants to get their site working again, I found that locking Vue Loader to the exact version mentioned in the error message fixed the problem. The old line (which the error implies is correct was) "vue-loader": "^15.9.5" which actually installed 15.9.6. The new line that works in real life is: `"vue-loader": "15.9.5"`

**Comment by luckydude777uz** [comment]
> So I can't exactly reproduce this but I have discovered a possibly related problem. Running mix with the production flags causes npm install to run with `NODE_ENV=production` which removes all dev dependencies.
> 
> Do you run into this if you just do `npm run dev` (no production flag)? Or `npx mix`?

npx mix works
